### ISSUE TITLE
jsTree and Browserify

### ISSUE THREAD

**Issue by Puffton** [vakata/jstree#1275]
Previously (v.3.1.1) I have used jsTree with Browserify as such: require('jstree'). Unfortunately, this behavior broke in v.3.2.0. A patch was released that fixed it, but it's back again in v3.2.1.

Browserify runs just fine and includes some jsTree code, but I get an error: Uncaught TypeError: $(...).jstree is not a function.

I can downgrade to 3.2.0 and it works, but I was momentarily tricked into thinking it worked with v3.2.1 as well, likely due to caching issues (I copy/pasted the trunk version of jsTree into node_modules at one point). Sorry about that. A clean download of all node_modules does show this error, rendering jsTree v3.2.1 unusable with Browserify.

**Comment by vakata** [comment]
Sorry, but jstree uses the default UMD pattern for jQuery modules. Maybe the problem is that you are not requiring jQuery, and only requiring jsTree? Or it could be that you have two jQuery copies in your code. The loading has not changed and the UMD pattern is still used, but keep in mind jstree will register itself inside the jQuery "namespace" - `jQuery`, so you can not access it directly when requiring.

Best regards,
Ivan

**Comment by Puffton** [comment]
Seems to have been a bug in Browserify. After upgrading I haven't had the issue.

**Comment by srinathh** [comment]
I ran into the same problem with jstree v3.2.1 on npm and browserify v.13.0.0 which is the latest on npm. Downgrading to jstree v3.2.0 again seems to work with the same version of browserify.

Here's how I'm importing and using jstree:
```
global.$ = require('jquery');
require('jstree');
$(function(){
    $("#treenode").jstree({
        "core" : {
           .........
```

**Comment by Puffton** [comment]
Yeah, I can confirm this is the case. Issue persists in jsTree 3.3.1 and has been pestering me for a year now. Can we re-open?

**Comment by vakata** [comment]
OK, but can you please provide a pull request for this - I have no idea how to fix this as I do not use browserify. If it is such an issue I will simply drop official support for browserify as I have no idea how to maintain this. All I did was use a common loaded, aside from that I can not help.

To sum up - can you provide an example that works?

**Comment by vakata** [comment]
OK, I did some testing, as stated on the site, I did this:
```
npm install -g browserify
```
Then I created a test project with a single HTML file:
```html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Test</title>

    <link rel="stylesheet" href="node_modules/jstree/dist/themes/default/style.css">
</head>
<body>
    <div id="jstree"></div>
    <script src="bundle.js"></script>
</body>
</html>
```
And a single js file:
```js
jquery = require('jquery');
require('jstree');

jquery('#jstree').jstree({ core : { data : ['Root'] } });
```

After that I executed:
```
browserify main.js -o bundle.js
```

After that I get a nicely rendered tree. I have never used browserify, so maybe this is way too simple, but it works for me. Please do tell which of the above did not work for you?

**Comment by Puffton** [comment]
Thanks for the quick feedback. Yeah, it seems tricky to debug. I'll try to upgrade to 3.2.1, then gradually downgrade to 3.2.0 file by file, until it works. Then at least we'll have isolated it to a specific file.

Not that I think it matters, but I'm on OS X 10.11, latest versions of npm, jstree and jQuery.

Other potential culprits: ReactJS, minifyify. Do you use any of them too, @srinathh?

I'm requiring it similarly, example:

```javascript
'use strict';
global.$ = require('jquery');
global.jQuery = global.$;
global.React = require('react');
global.ReactDOM = require('react-dom');
global.PureRenderMixin = require('react-addons-pure-render-mixin');

global.ReactRouter = require('react-router');
global.hashHistory = global.ReactRouter.hashHistory;
global.IndexRoute = ReactRouter.IndexRoute;
global.Router = ReactRouter.Router;
global.Route = ReactRouter.Route;
global.Link = ReactRouter.Link;
global.History = ReactRouter.History;

require('ion-rangeslider');
require('jstree');
```

**Comment by Puffton** [comment]
To reproduce the issue I tried to remove jsTree and then download it again using npm, but it didn't cause issues. What did cause issues was when I removed my entire node_modules dir, and downloaded all of my dependencies again. The order in which the deps are downloaded affects the bundled JavaScript-file generated by Browserify. The issue could be more pronounced when you have more deps, my project has about 20 npm deps.

The solution is, after downloading all js dependencies, delete jstree in node_modules and re-download it using npm. This is just a theory: Normally when a component depends on another, it has it included in package.json. Since jsTree depends indirectly on jQuery (without specifying it), it is possible that, when it is jQuery, it is somehow included by Browserify after jsTree (even though it is included above in the require.js file). So, when I'm trying to use jsTree, I haven't defined jQuery.

That's not really a bug though, I wouldn't want each jQuery plugin to download a separate instance of jQuery, but it's good to be aware of.
